File: TwoPointers/python/compress.py
def compress(s: str) -> str:
    if not s:
        return ""

    parts: list[str] = []
    i = 0
    while i < len(s):
        j = i + 1
        while j < len(s) and s[j] == s[i]:
            j += 1
        count = j - i
        if count == 1:
            parts.append(s[i])
        else:
            parts.append(str(count))
            parts.append(s[i])
        i = j

    return "".join(parts)

File: TwoPointers/go/compress.go
package twopointers

import (
	"strconv"
	"strings"
)

// Compress performs run-length encoding on s.
func Compress(s string) string {
	var builder strings.Builder
	i := 0
	for i < len(s) {
		j := i + 1
		for j < len(s) && s[j] == s[i] {
			j++
		}
		count := j - i
		if count == 1 {
			builder.WriteByte(s[i])
		} else {
			builder.WriteString(strconv.Itoa(count))
			builder.WriteByte(s[i])
		}
		i = j
	}
	return builder.String()
}
